BSP/STM32F413/Src/BSP_PWM.c

```diff
@@ -0,0 +1,170 @@
+    GPIO_PinAFConfig(GPIOB, GPIO_PinSource14, GPIO_AF_TIM12);
+    GPIO_PinAFConfig(GPIOB, GPIO_PinSource15, GPIO_AF_TIM12);
+
+    //All pins will be initialized for alternate function
+    GPIO_INIT_STRUCT.GPIO_Mode = GPIO_Mode_AF;
+    GPIO_INIT_STRUCT.GPIO_Speed = GPIO_Speed_2MHz;
+    GPIO_INIT_STRUCT.GPIO_PuPd = GPIO_PuPd_DOWN;
+    GPIO_INIT_STRUCT.GPIO_OType = GPIO_OType_PP;
+    //Initialize Pin PC6, PC7
+    GPIO_INIT_STRUCT.GPIO_Pin = GPIO_Pin_6 | GPIO_Pin_7;
+    GPIO_Init(GPIOC, &GPIO_INIT_STRUCT);
+    //Initialize Pin PB14, PB15
+    GPIO_INIT_STRUCT.GPIO_Pin = GPIO_Pin_14 | GPIO_Pin_15 | GPIO_Pin_0;
+    GPIO_Init(GPIOB, &GPIO_INIT_STRUCT);
+
+    //Configure Time base unit
+	TIMER_INIT_STRUCT.TIM_Period = PWM_PERIOD;
+  	TIMER_INIT_STRUCT.TIM_Prescaler = 0x0000;
+	TIMER_INIT_STRUCT.TIM_ClockDivision = TIM_CKD_DIV1;
+	TIMER_INIT_STRUCT.TIM_CounterMode = TIM_CounterMode_Up;
+	TIMER_INIT_STRUCT.TIM_RepetitionCounter = 0x0000;
+    TIM_TimeBaseInit(TIM3, &TIMER_INIT_STRUCT);
+    TIM_TimeBaseInit(TIM8, &TIMER_INIT_STRUCT);
+    TIM_TimeBaseInit(TIM12, &TIMER_INIT_STRUCT);
+
+    //Fill in TIM_OCInitStruct with desired parameters
+	TIMER_OC_STRUCT.TIM_OCMode = TIM_OCMode_PWM1;
+	TIMER_OC_STRUCT.TIM_OutputState = TIM_OutputState_Enable;
+	TIMER_OC_STRUCT.TIM_OutputNState = TIM_OutputNState_Disable;
+	TIMER_OC_STRUCT.TIM_Pulse = 0; //start with everything off
+	TIMER_OC_STRUCT.TIM_OCPolarity = TIM_OCPolarity_High;
+	TIMER_OC_STRUCT.TIM_OCNPolarity = TIM_OCPolarity_High;
+	TIMER_OC_STRUCT.TIM_OCIdleState = TIM_OCIdleState_Reset;
+	TIMER_OC_STRUCT.TIM_OCNIdleState = TIM_OCNIdleState_Reset;
+	//Configure all channels with the output compare
+    TIM_OC3Init(TIM3, &TIMER_OC_STRUCT);
+    TIM_OC1Init(TIM8, &TIMER_OC_STRUCT);
+    TIM_OC2Init(TIM8, &TIMER_OC_STRUCT);
+    TIM_OC1Init(TIM12, &TIMER_OC_STRUCT);
+    TIM_OC2Init(TIM12, &TIMER_OC_STRUCT);
+
+    //Enable the output compare preload on all channels
+    TIM_OC3PreloadConfig(TIM3, TIM_OCPreload_Enable);
+    TIM_OC1PreloadConfig(TIM8, TIM_OCPreload_Enable);
+    TIM_OC2PreloadConfig(TIM8, TIM_OCPreload_Enable);
+    TIM_OC1PreloadConfig(TIM12, TIM_OCPreload_Enable);
+    TIM_OC2PreloadConfig(TIM12, TIM_OCPreload_Enable);
+
+    TIM_ARRPreloadConfig(TIM12, ENABLE);
+
+    TIM_CtrlPWMOutputs(TIM8, ENABLE);
+
+    //Enable the TIM8 counter
+    TIM_Cmd(TIM3, ENABLE);
+    TIM_Cmd(TIM8, ENABLE);
+    TIM_Cmd(TIM12, ENABLE);
+    //END OF PWM Init method
+
+
+    //THIS IS EXTRA CODE THAT HAS NO HOME, SO I'M PUTTING IT IN THE PWM FILE FOR NOW.
+    //This deals with setting up the input pin for the contactor
+    //setup the input pin
+    GPIO_InitTypeDef GPIO_C1Init;
+	GPIO_C1Init.GPIO_Pin = GPIO_Pin_1; //input pin is gpio B1
+    GPIO_C1Init.GPIO_Mode = GPIO_Mode_IN;
+    GPIO_C1Init.GPIO_Speed = GPIO_Speed_2MHz;
+    GPIO_C1Init.GPIO_PuPd = GPIO_PuPd_DOWN;
+    GPIO_C1Init.GPIO_OType = GPIO_OType_PP;
+    GPIO_Init(C1_PORT, &GPIO_C1Init);
```

Comment:
unless this is for the other two contactors on the line we tested last workday. Something is off in the initialization for the fan contactor but the rest of it looks good. We should be able to test fans and fan contactor together

Reply:
This should be fixed in another PR. After you debug the Contactors stuff.